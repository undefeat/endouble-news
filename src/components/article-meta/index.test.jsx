import React from 'react';
import shallow from 'react-test-renderer/shallow';
import ArticleMeta from '.';

describe('ArticleMeta', () => {
    it('should render Loader when fetching', () => {
        Date.now = jest.fn(() => 1549722223384);

        const tree = shallow
            .createRenderer()
            .render(
                <ArticleMeta
                    source="CNBC"
                    url="https://www.cnbc.com/2019/02/05/apple-homepod-smart-speaker-market-share.html"
                    publishedAt="2019-02-05T14:06:38Z"
                />,
            );

        expect(tree).toMatchSnapshot();
    });
});
