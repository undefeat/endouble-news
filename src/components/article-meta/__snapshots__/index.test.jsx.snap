// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleMeta should render Loader when fetching 1`] = `
<div
  className="article-meta"
>
  <a
    aria-label="Source URL"
    className="article-meta__source-link"
    href="https://www.cnbc.com/2019/02/05/apple-homepod-smart-speaker-market-share.html"
    rel="noopener noreferrer"
    target="_blank"
  >
    CNBC
  </a>
  <span
    className="article-meta__delimiter"
  >
    |
  </span>
  <time
    className="article-meta__date"
    dateTime="2019-02-05T14:06:38Z"
    title="2/5/2019, 4:06:38 PM"
  >
    4 days ago
  </time>
</div>
`;
